lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/file/download/v2/DownloadListDaoImplTest.java

```diff
@@ -322,17 +352,724 @@ public void testClearDownloadListWithMultipleUsers() {
+		List<Node> files = createFileHierarchy(numberOfProject, foldersPerProject, filesPerFolder);
+		assertEquals(6, files.size());
+		List<Long> fileIds = files.stream().map(n -> KeyFactory.stringToKey(n.getId())).collect(Collectors.toList());
+
+		List<DownloadListItem> userOneItems = Arrays.asList(
+				new DownloadListItem().setFileEntityId(files.get(0).getId()).setVersionNumber(1L),
+				new DownloadListItem().setFileEntityId(files.get(1).getId()).setVersionNumber(2l),
+				new DownloadListItem().setFileEntityId(files.get(3).getId()).setVersionNumber(null),
+				new DownloadListItem().setFileEntityId(files.get(4).getId()).setVersionNumber(1L),
+				new DownloadListItem().setFileEntityId(files.get(5).getId()).setVersionNumber(null));
+
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, userOneItems);
+
+		List<DownloadListItemResult> expected = downloadListDao.getDownloadListItems(userOneIdLong,
+				userOneItems.get(1), userOneItems.get(3), userOneItems.get(4));
+		
+		// grant access to a sub-set of the files
+		List<Long> subSet = Arrays.asList(fileIds.get(1), fileIds.get(4), fileIds.get(5));
+
+		List<Sort> sort = Arrays.asList(new Sort().setField(SortField.synId).setDirection(SortDirection.ASC));
+		Long limit = 100L;
+		Long offset = 0L;
+		// call under test
+		List<DownloadListItemResult> result = downloadListDao.getFilesAvailableToDownloadFromDownloadList(l -> subSet,
+				userOneIdLong, sort, limit, offset);
+		assertEquals(expected, result);
+	}
+	
+	@Test
+	public void testGetFilesAvailableToDownloadFromDownloadListWithLimitOffset() {
+		int numberOfProject = 2;
+		int foldersPerProject = 1;
+		int filesPerFolder = 3;
+		List<Node> files = createFileHierarchy(numberOfProject, foldersPerProject, filesPerFolder);
+		assertEquals(6, files.size());
+
+		List<DownloadListItem> userOneItems = Arrays.asList(
+				new DownloadListItem().setFileEntityId(files.get(0).getId()).setVersionNumber(1L),
+				new DownloadListItem().setFileEntityId(files.get(1).getId()).setVersionNumber(2l),
+				new DownloadListItem().setFileEntityId(files.get(3).getId()).setVersionNumber(null),
+				new DownloadListItem().setFileEntityId(files.get(4).getId()).setVersionNumber(1L),
+				new DownloadListItem().setFileEntityId(files.get(5).getId()).setVersionNumber(null));
+
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, userOneItems);
+
+		List<DownloadListItemResult> expected = downloadListDao.getDownloadListItems(userOneIdLong, userOneItems.get(3),
+				userOneItems.get(4));
+
+		List<Sort> sort = Arrays.asList(new Sort().setField(SortField.synId).setDirection(SortDirection.ASC));
+		Long limit = 2L;
+		Long offset = 3L;
+		// call under test
+		List<DownloadListItemResult> result = downloadListDao.getFilesAvailableToDownloadFromDownloadList(l -> l,
+				userOneIdLong, sort, limit, offset);
+		assertEquals(expected, result);
+	}
+	
+	@Test
+	public void testGetFilesAvailableToDownloadFromDownloadListWithOrderByProjectName() {
```

Comment:
We should probably add a test for at least the file name ordering to make that works.